tests/rptest/utils/si_utils.py

```diff
@@ -242,6 +242,43 @@ def get_on_disk_size_per_ntp(chk):
     return size_bytes_per_ntp
 
 
+def get_expected_ntp_restored_size(nodes_segments_report: dict[str,
+                                                               dict[str,
+                                                                    (str,
+                                                                     int)]],
+                                   retention_policy: int):
+    """ Get expected retestored ntp disk size
+    We expect that redpanda will restore max
+    amount of segments with total size less
+    than retention_policy
+    """
+    size_bytes_per_ntp = {}
+    segments_sizes_per_ntp = {}
+    for _node, node_segments_reports in nodes_segments_report.items():
```

Comment:
`node_segments_report` shadows the parameter and should probably be renamed

Reply:
parameter has data of multiple nodes, so its name is
`nodes_segments_report`

Reply:
Maybe `for _node, report in nodes_segments_report.items()`

Even if the variables don't literally shadow each other, it's a bit too subtle to have them differ by just one `s` in the middle.

Reply:
changed to report